internal/cli/atlas/privateendpoints/gcp/watch.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type WatchOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	id    string
+	store store.PrivateEndpointDescriber
+}
+
+func (opts *WatchOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.PrivateEndpoint(opts.ConfigProjectID(), provider, opts.id)
+	if err != nil {
+		return false, err
+	}
+	return result.Status == "WAITING_FOR_USER" || result.Status == "FAILED", nil
```

Comment:
These are the available status to use here, you probably need to use `AVAILABLE` instead of `WAITING_FOR_USER`

https://docs.atlas.mongodb.com/reference/api/private-endpoints-service-get-one/
![Screenshot 2022-01-18 at 09 20 56](https://user-images.githubusercontent.com/5663078/149907894-8e829856-ee25-4d4d-8e08-f8f601a8588f.png)

Reply:
Thanks, that makes sense. Got confused by status doc and thought it was the same for all providers.

Reply:
Have you tested this command with a new private endpoint? I am wondering if here we should use `INITIATING` instead of `AVAILABLE`

Reply:
From the doc description, I understood we're waiting for it to be available. The test:
- Note: The watch only stopped after a while which was the time the endpoint got available
```
➜  mongocli git:(CLOUDP-108869) ✗ ./bin/mongocli atlas privateEndpoints gcp create --projectId 61dc598fd1f07c7b37ba425d --region EASTERN_US
GCP private endpoint '61e690f4c07ec4093753aa54' created.
➜  mongocli git:(CLOUDP-108869) ✗ mongocli atlas privateEndpoints gcp watch --projectId 61dc598fd1f07c7b37ba425d 61e690f4c07ec4093753aa54

GCP Private endpoint changes completed.
➜  mongocli git:(CLOUDP-108869) ✗
➜  mongocli git:(CLOUDP-108869) ✗
➜  mongocli git:(CLOUDP-108869) ✗
➜  mongocli git:(CLOUDP-108869) ✗ mongocli atlas privateEndpoints gcp list --projectId 61dc598fd1f07c7b37ba425d
ID                         REGION       STATUS      ERROR
61e688267969f71a32b18671   CENTRAL_US   AVAILABLE
61e690f4c07ec4093753aa54   EASTERN_US   AVAILABLE
➜  mongocli git:(CLOUDP-108869) ✗
```